server/middlewares/middleware.js

```diff
@@ -0,0 +1,59 @@
+const TodoController = require("../controllers/todoController")
+const {
+  cekToken
+} = require("../helpers/jwt")
+const {
+  User,
+  Todo
+} = require("../models")
+
+
+const authentication = (req, res, next) => {
+  try {
+    let decoded = cekToken(req.headers.access_token)
+    User.findOne({
+      where: {
+        email: decoded.email
+      }
+    })
+    .then(data => {
+      if (!data) {
+        res.status(401).json({message: "please login first"})
+      } else {
+        req.user = data
+        next()
+      }
+    })
+    .catch(err => {
+      res.status(500).json({message: err.message})
+    })
+    
+  } catch (err) {
+    res.status(400).json({message: err.message})
+  }
+}
+const authorization = (req, res, next) => {
+  let id = req.params.id
+  Todo.findByPk(id)
```

Comment:
disini ditambahin juga if datanya ada ya mas, jadi misalnya gk ada langsung di lempar ke error handler data not found

Reply:
oh, oke kak. Sekarang udh ditambahin case kalo datanya ga ada kak